common.go

```diff
@@ -40,13 +36,7 @@ func NullString() *string {
 	return output
 }
 
-func NewString(value string) *string {
```

Comment:
Should we keep this around - mark them as deprecated and under the hood of them use `RefOf`

This will give opslevel-go upgraders a warning about switching off this for this new paradigm that will be just introduced in the 2024 release?

Reply:
@rocktavious yes I think that's a good idea. To be extra annoying we could even `log.Debug()` (or info? that might break stuff) about it. I don't think `ref.Of()` is something that comes to mind intuitively and would be a pretty big shock.